File: frontend/src/api/risk.ts
/** /api/kill-switch / /api/promotion/* — 체크리스트 #75. */
import { apiFetch } from "./client";

export interface KillSwitchRequest {
  active: boolean;
  reason?: string;
}

export interface KillSwitchResponse {
  active: boolean;
  reason: string;
}

export function setKillSwitch(req: KillSwitchRequest, adminToken: string) {
  return apiFetch<KillSwitchResponse>("/api/kill-switch", {
    method: "POST",
    body: JSON.stringify(req),
    adminToken,
  });
}

export interface PromotionGateMetrics {
  passed: boolean;
  from_mode: string;
  to_mode: string;
  reason: string;
}

export function fetchPaperGate() {
  return apiFetch<PromotionGateMetrics>("/api/promotion/paper-gate");
}

export function fetchShadowGate() {
  return apiFetch<PromotionGateMetrics>("/api/promotion/shadow-gate");
}

File: backend/app/api/risk.py
"""Risk 라우터 — /api/kill-switch + /api/promotion/*."""
from dataclasses import asdict
from fastapi import APIRouter, Depends
from pydantic import BaseModel

from app.governance.promotion_gates import check_paper_gate, check_shadow_gate
from .deps import gateway, verify_admin

router = APIRouter()


class KillSwitchRequest(BaseModel):
    active: bool
    reason: str = ""


class PromotionMetrics(BaseModel):
    sharpe: float = 0.0
    max_drawdown_pct: float = 0.0
    win_rate_pct: float = 0.0
    total_trades: int = 0
    weeks_run: int = 0
    shadow_weeks: int = 0
    p95_latency_ms: float = 0.0
    failure_drills_done: int = 0


@router.post("/api/kill-switch")
def kill_switch(body: KillSwitchRequest, _=Depends(verify_admin)):
    gateway.kill_switch(body.active, body.reason)
    return {"kill_switch": body.active, "reason": body.reason}


@router.post("/api/promotion/paper-gate")
def paper_gate(metrics: PromotionMetrics, _=Depends(verify_admin)):
    return asdict(check_paper_gate(metrics.model_dump()))


@router.post("/api/promotion/shadow-gate")
def shadow_gate(metrics: PromotionMetrics, _=Depends(verify_admin)):
    return asdict(check_shadow_gate(metrics.model_dump()))
